launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -50,12 +52,31 @@ def method_2():
             print(wait_for_topics.topics_received()) # Should be {'topic_1', 'topic_2'}
             print(wait_for_topics.messages_received('topic_1')) # Should be [message_1, ...]
             wait_for_topics.shutdown()
+
+        # Method3, calling a callback function before the wait. The callback function takes
+        # the WaitForTopics object as the first argument. Any additional arguments has
+        # to be passed to the wait(*args, **kwargs) method directly.
+        def callback_function(node, arg=""):
+            node.get_logger().info('Callback function called with argument: ' + arg)
+
+        def method_3():
+            topic_list = [('topic_1', String), ('topic_2', String)]
+            wait_for_topics = WaitForTopics(topic_list, timeout=5.0, callback=callback_function)
+            # The callback function will be called inside the wait() method after the
+            # subscribers are created and before the publishers are connected.
+            assert wait_for_topics.wait("Hello World!")
+            print('Given topics are receiving messages !')
+            wait_for_topics.shutdown()
     """
 
-    def __init__(self, topic_tuples, timeout=5.0, messages_received_buffer_length=10) -> None:
+    def __init__(self, topic_tuples, timeout=5.0, messages_received_buffer_length=10,
+                 callback=None):
```

Comment:
Just to keep the return type:

```suggestion
                 callback=None) -> None:
```